common/workunit/workunit.cpp

```diff
@@ -2946,6 +3081,9 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 return ret;
             }
         };
+        CWorkUnitsPagerHelper helper;
+        bool hasOrFilter = false;
+        CQueryOrFilter orFilter;
```

Comment:
This local variable is passed to the constructor of CWorkUnitsPager which saves the pointer in the object, and the object then returned.  That is surely wrong.